### PR TITLE
[DRAFT] fix misordered upsert keys

### PR DESCRIPTION
**Pull request by umanwizard**

### CODE REVIEW

File: src/dataflow/src/storage/render/upsert.rs [hunk 1]
```diff
@@ -552,35 +552,43 @@ where
...
+            row_packer.push(value)
+        }
     }
-    // Finally, push any columns after the last key index
-    row_packer.extend(values);
-
     row_buf.clone()
 }
 
 /// `rehydrate` uses information from the source description to find which indexes in the row
```
Comment (umanwizard) [inline src/dataflow/src/storage/render/upsert.rs:566]:
```suggestion
/// `rehydrate` uses information from the source description to find which indices in the row
```

might as well be consistent here

File: src/dataflow/src/storage/render/upsert.rs [hunk 2]
```diff
@@ -643,7 +651,7 @@ mod tests {
     fn test_rehydrate_thin_multiple() {
         let mut packer = Row::default();
 
-        let key_indices = vec![2, 4];
+        let key_indices = vec![(2, 0), (4, 0)].into_iter().collect();
```
Comment (umanwizard) [inline src/dataflow/src/storage/render/upsert.rs:654]:
```suggestion
        let key_indices = vec![(2, 0), (4, 1)].into_iter().collect();
```

**Comment by guswynn** [comment]
superceded by https://github.com/MaterializeInc/materialize/pull/12727

**Comment by guswynn** [comment]
``

**Pull request closed without merging**